main.cpp

```diff
@@ -0,0 +1,117 @@
+#include "multiphasesorter.h"
+
+#include <fstream>
+#include <random>
+
+/// Генерирует файл с рандомными значениями в диапазоне от -maxNumberValue до maxNumberValue
+bool createFileWithRandomNumbers(const std::string &fileName,
+                                 const int numbersCount,
+                                 const int maxNumberValue)
+{
+    std::ofstream outFile(fileName);
+
+    if (!outFile) {
+        std::cerr << "Error: Could not open file " << fileName << " for writing.\n";
+        return false;
+    }
+
+    std::mt19937 rand(::time(0));
+
+    for (int i = 0; i < numbersCount; ++i) {
+        int randomNumber = std::rand() % (2 * maxNumberValue + 1) - maxNumberValue;
+        outFile << randomNumber << " ";
+    }
+
+    outFile.close();
+
+    if (!outFile) {
+        std::cerr << "Error: Failed to write to file " << fileName << ".\n";
+        return false;
+    }
+
+    return true;
+}
+
+bool isFileContainsSortedArray(const std::string &fileName)
+{
+    std::ifstream inFile(fileName);
+    if (!inFile) {
+        std::cerr << "Error: Could not open file " << fileName << " for reading.\n";
+        return false;
+    }
+
+    int previousNumber;
+    bool firstNumber = true;
+    int currentNumber;
+
+    while (inFile >> currentNumber) {
+        if (firstNumber) {
+            previousNumber = currentNumber;
+            firstNumber = false;
+            continue;
+        }
+
+        if (currentNumber < previousNumber) {
+            return false;
+        }
+
+        previousNumber = currentNumber;
+    }
+
+    return true;
+}
```

Comment:
Здесь можно чуть упростить код, если до цикла считать данные в `previousNumber`. Если файл пустой или содержит одно число, функция не зайдёт в цикл и вернёт `true`.
```suggestion
    int previousNumber;
    int currentNumber;
    inFile >> previousNumber;

    while (inFile >> currentNumber) {
        if (currentNumber < previousNumber) {
            return false;
        }

        previousNumber = currentNumber;
    }

    return true;
}
```

Reply:
Согласен